src/RelayPricer.ts

```diff
@@ -0,0 +1,28 @@
+import { SourceApi } from './types/RelayPricer';
+
+export default class RelayPricer {
+    private readonly sourceApi: SourceApi;
+
+    constructor(sourceApi: SourceApi) {
+        this.sourceApi = sourceApi;
+    }
+
+    async getPrice(
+        sourceCurrency: string,
+        targetCurrency: string,
+        intermediateCurrency?: string
+    ): Promise<number> {
+        const intermediary = intermediateCurrency
+            ? intermediateCurrency
+            : 'USD';
+
+        const [sourcePrice, targetPrice] = await Promise.all([
+            this.sourceApi.query(sourceCurrency, intermediary),
+            this.sourceApi.query(targetCurrency, intermediary)
+        ]);
+        if (sourcePrice === 0 || targetPrice === 0) {
+            throw new Error('price cannot be zero');
+        }
```

Comment:
This message is being returned to the caller but the caller is not the source of this 0. I think this message can be improved.

Reply:
Furthermore, to store the error message we could use a constant which will be reused in the tests

Reply:
updated

Reply:
how about 
```
sourceApi `Currency conversion for pair ${sourceCurrency}:${targetCurrency} not found in current source api`);
```
or if we were to add a property to SourceApi interface: 
```
sourceApi `Currency conversion for pair ${sourceCurrency}:${targetCurrency} not found in ${this.sourceApi.name}`);
```
or further, if we were to make it accept a number of source apis: 
```
sourceApi `Currency conversion for pair ${sourceCurrency}:${targetCurrency} not found in any currently used apis. Use `listSourceApis` to list these.`);
```

which brings me to a new question: "SourceApi"... is that a meaningful name? Perhaps better would be "PriceApi" or "ExchangeApi" or "CurrencyConversionApi"? That would describe its job, whereas "source" is a bit vague.

Reply:
I vote for ExchangeApi